machine_learning/fp_growth.py

```diff
@@ -0,0 +1,335 @@
+"""
```

Comment:
Please change the filename to `frequent_pattern_growth.py` so that it is self-documenting to readers.